docs/concepts/glossary.md

```diff
@@ -145,10 +188,18 @@ Graphsync is an alternative content replication protocol under discussion, simil
 
 ## H
 
+### HAMT-sharding
+
+The sharding technique used for [sharding](#sharding) big UnixFS directories. It leverages properties of hash array mapped tries (HAMT). [More about HAMT](https://en.wikipedia.org/wiki/Hash_array_mapped_trie).
+
 ### Hash
 
 A Cryptographic Hash is a function that takes some arbitrary input (content) and returns a fixed-length value. The exact same input data will always generate the same hash as output. There are numerous hash algorithms. [More about Hash](hashing.md)
 
+### Hole punching
+
+A generic term for NAT traversal when port forwarding is not possible. [See DCUtR](#dcutr)
```

Comment:
not exactly right, it is a specific technique that relies on coordinated simultaneous connections.

Reply:
```suggestion
A generic term for NAT / firewall traversal used when port forwarding is not possible. [See DCUtR](#dcutr)
```

Nit pick.

@vyzo I am not sure I understand your comment.

Reply:
I mean it is a technique, not a totally generic term

Reply:
the texniq is generic however

Reply:
@vyzo would this work?

```suggestion
A  technique for NAT / firewall traversal that relies on coordinated simultaneous connections.  Used when port forwarding is not possible. [See DCUtR](#dcutr)
```

Reply:
yeah, thats quite good.